python/hail/tests.py

```diff
@@ -773,3 +774,45 @@ def test_concordance(self):
 
         glob, samples, variants = bn1.concordance(bn2)
         self.assertEqual(samples.sample_annotations[samples.sample_ids[0]].concordance, glob)
+
+    def test_hdfs_files(self):
+        data = ['foo', 'bar', 'baz']
+        data.extend(map(str, range(100)))
+
+        with hdfs_write('/tmp/test_out.txt') as f:
+            for d in data:
+                f.write(d)
+                f.write('\n')
+
+        with hdfs_read('/tmp/test_out.txt') as f:
+            data2 = [line.strip() for line in f]
+
+        self.assertEqual(data, data2)
+
+        with hdfs_write('/tmp/test_out.txt.gz') as f:
+            for d in data:
+                f.write(d)
+                f.write('\n')
+
+        with hdfs_read('/tmp/test_out.txt.gz') as f:
+            data3 = [line.strip() for line in f]
+
+        self.assertEqual(data, data3)
+
+        hdfs_copy('/tmp/test_out.txt.gz', '/tmp/test_out.copy.txt.gz')
+
+        with hdfs_read('/tmp/test_out.copy.txt.gz') as f:
+            data4 = [line.strip() for line in f]
+
+        self.assertEqual(data, data4)
+
+        with open('/tmp/numbers.txt', 'w') as f:
+            for i in range(1000):
+                f.write(str(i))
+                f.write('\n')
+
+        with hdfs_read('/tmp/numbers.txt', buffer_size=75) as f:
```

Comment:
How big was your binary test file?  Do we want to include that, too?  The encoding business makes me nervous (and might break when we, say, start testing on python 3).